### ISSUE TITLE
can raise RequestTimeout on connection timeout

### ISSUE THREAD

**Issue by belm0** [theelous3/asks#127]
in request():
```
sock = await timeout_manager(
     connection_timeout, self._grab_connection, url)
```
so this is establishing a connection, not making the request.  Yet timeout_manager() will raise RequestTimeout.

Overall it would be helpful if exception cases are documented for request().

**Comment by theelous3** [comment]
Fair point. I lazily reused that for both the connection and the request, rather than give them a base class.

I'll sort this and slap it in the docs :)

**Comment by PurpleMyst** [comment]
This was resolved in ced7a74a

**Comment by belm0** [comment]
> This was resolved in ced7a74

I don't think so, since that commit predates the problem report by a year.

**Comment by theelous3** [comment]
Yeah no, @belm0 was asking for two separate possible exceptions:

`ConnectionTimeout` on failed connection attempts.
`RequestTimeout` on successful connects where the server didn't respond to the https request on time.
